gpu-miner.c

```diff
@@ -72,34 +72,62 @@ double grindNonces(size_t global_item_size) {
 		return 0;
 	}
 
-	// Copy input data to the memory buffer
-	ret = clEnqueueWriteBuffer(command_queue, blockHeadermobj, CL_TRUE, 0, 80 * sizeof(uint8_t), blockHeader, 0, NULL, NULL);
-	if (ret != CL_SUCCESS) { printf("failed to write to blockHeadermobj buffer: %d\n", ret); exit(1); }
-	ret = clEnqueueWriteBuffer(command_queue, headerHashmobj, CL_TRUE, 0, 32 * sizeof(uint8_t), headerHash, 0, NULL, NULL);
-	if (ret != CL_SUCCESS) { printf("failed to write to headerHashmobj buffer: %d\n", ret); exit(1); }
-	ret = clEnqueueWriteBuffer(command_queue, targmobj, CL_TRUE, 0, 32 * sizeof(uint8_t), target, 0, NULL, NULL);
-	if (ret != CL_SUCCESS) { printf("failed to write to targmobj buffer: %d\n", ret); exit(1); }
-
-	// Execute OpenCL kernel as data parallel
-	size_t local_item_size = 256;
-	global_item_size -= global_item_size % 256;
-	ret = clEnqueueNDRangeKernel(command_queue, kernel, 1, NULL, &global_item_size, &local_item_size, 0, NULL, NULL);
-	if (ret != CL_SUCCESS) { printf("failed to start kernel: %d\n", ret); exit(1); }
-
-	// Copy result to host
-	ret = clEnqueueReadBuffer(command_queue, headerHashmobj, CL_TRUE, 0, 32 * sizeof(uint8_t), headerHash, 0, NULL, NULL);
-	if (ret != CL_SUCCESS) { printf("failed to read header hash from buffer: %d\n", ret); exit(1); }
-	ret = clEnqueueReadBuffer(command_queue, nonceOutmobj, CL_TRUE, 0, 8 * sizeof(uint8_t), nonceOut, 0, NULL, NULL);
-	if (ret != CL_SUCCESS) { printf("failed to read nonce from buffer: %d\n", ret); exit(1); }
-
-	// Did we find one?
-	if (memcmp(headerHash, target, 8) < 0) {
-		// Copy nonce to block
-		memcpy(block+32, nonceOut, 8);
-		submit_block(curl, block, blocklen);
-		blocks_mined++;
+	// Check for target corruption
+	if (target[0] != 0 || target[1] != 0 || target[2] != 0 || target[3] != 0) {
+		if (target_corrupt_flag) {
+			return -1;
+		}
+		target_corrupt_flag = 1;
+		printf("Received corrupt target from Sia\n");
+		printf("Usually this resolves itself within a minute or so\n");
+		printf("If it happens frequently trying increasing seconds per iteration\n");
+		printf("e.g. \"./gpu-miner -s 3 -c 200\"\n");
+		printf("Waiting for problem to be resolved...");
+		fflush(stdout);
```

Comment:
in my tests it printed once and then not again, never resolved itself. Why was that?